File: Curso Java Com POO/Java com POO/Aula14e15(Exercicio Pratico 2)/src/aula14e15/Gafanhoto.java
package aula14e15;

//Classe Gafanhoto estendendo classe Pessoa
public class Gafanhoto extends Pessoa {
    //Atributos
    private String login;
    private int totAssistido;
    
    
    //Construtor
    public Gafanhoto(String nome, int idade, String sexo, String login) {
        super(nome, idade, sexo);
        this.login = login;
        this.totAssistido = 0;
    }

    
    //Método Sobreposto da classe Pessoa
    @Override
    protected void ganharExp() {
        
    }
    
    //Métodos
    public void viuMaisUm() {
        
    }
    
    
    //Método toString
    @Override
    public String toString() {
        return "Gafanhoto{" 
                + super.toString() 
                + " -> login=" + login 
                + ", totAssistido=" + totAssistido 
                + '}';
    }
    
    
    //Getters e Setters
    //login
    public String getLogin() {
        return login;
    }
    public void setLogin(String login) {
        this.login = login;
    }

    //totAssistido
    public int getTotAssistido() {
        return totAssistido;
    }
    public void setTotAssistido(int totAssistido) {
        this.totAssistido = totAssistido;
    }
    
}

File: Curso Programação Orientada a Objetos/PHP com POO/Aula14e15-ExercicioPratico2/Gafanhoto.php
<?php

require_once 'Pessoa.php';

//Classe Gafanhoto estendendo classe Pessoa
class Gafanhoto extends Pessoa{
    //Atributos
    private $login;
    private $totAssistido;

    //Construtor
    public function __construct($nome, $idade, $sexo, $login) {
        parent::__construct($nome, $idade, $sexo); //semelhante ao: super() em java
        $this->login = $login;
        $this->totAssistido = 0;
    }

        
    //Método Sobreposto da classe Pessoa
    protected function ganharExp($x) { //x -> experiência ganha 
        $this->setExperiencia($this->getExperiencia() + $x);
        //$this->experiencia += $x;
    }
    
    //Métodos
    public function viuMaisUm() {
        $this->setTotAssistido($this->getTotAssistido() + 1);
        //$this->totAssistido ++;
    }
    
    
    //Getters 
    public function getLogin() {
        return $this->login;
    }

    public function getTotAssistido() {
        return $this->totAssistido;
    }

    //Setters
    public function setLogin($login) {
        $this->login = $login;
    }

    public function setTotAssistido($totAssistido) {
        $this->totAssistido = $totAssistido;
    }


    
}
